elements/lisk-cryptography/src/sign.ts

```diff
@@ -141,11 +141,15 @@ export const signDataWithPassphrase = (
 	data: Buffer,
 	passphrase: string,
 ): string => {
-	const { privateKeyBytes } = getPrivateAndPublicKeyBytesFromPassphrase(
-		passphrase,
-	);
+	if (data && passphrase) {
```

Comment:
if it requires the check, then the input type is wrong, but i think in this case, the usage of the function is wrong.
Otherwise, to make it extream, u need to check if it's Buffer or string too

Reply:
buffer or string is checked in the hash function.